astropy/io/ascii/ecsv.py

```diff
@@ -136,12 +134,7 @@ def get_cols(self, lines):
         try:
             header = meta.get_header_from_yaml(lines)
         except ImportError as exc:
-            if 'PyYAML package is required' in str(exc):
-                warnings.warn("file looks like ECSV format but PyYAML is not installed "
-                              "so it cannot be parsed as ECSV",
-                              AstropyWarning)
-            raise core.InconsistentTableError('unable to parse yaml in meta header'
-                                              ' (PyYAML package is required)')
+            raise core.InconsistentTableError('unable to parse yaml in meta header')
```

Comment:
There isn't a non-yaml test for this error. Remove error?

Reply:
Tricky one. In principle, the yaml could get mangled somehow, and then the parsing would fail. When I first reviewed, I thought it is fine to raise a clearer message, but looking at it again, my tendency would indeed be to remove the error and just lead the `get_header...` fail, since this really should be rare.

Reply:
Done!